acl tests succeed in kernel acl check but failed in user space acl check

**What happened**:
root@bench-01:~/juicefs-jiefeng# ./juicefs format sqlite3://test.db acl --enable-acl --trash-days 0
root@bench-01:~/juicefs-jiefeng# ./juicefs mount sqlite3://test.db /tmp/jfs --non-default-permission -d
root@bench-01:~/juicefs# python3 .github/scripts/hypo/fsrand2_test.py -k test_acl_913
setup_logger ./log1
setup_logger ./log2
__init__
duration is 0.036437034606933594
2024-03-14 20:00:07,612 - INFO - do_create_file /tmp/fsrand/aaaa with mode w succeed
2024-03-14 20:00:07,618 - INFO - do_create_file /tmp/jfs/fsrand/aaaa with mode w succeed
2024-03-14 20:00:07,618 - INFO - run_cmd: sudo -u root setfacl       -m u:user1:-,g:root:-,o::- /tmp/fsrand/
run_cmd:getfacl /tmp/fsrand/
2024-03-14 20:00:07,624 - INFO - do_set_acl /tmp/fsrand/ with u:user1:-,g:root:-,o::- succeed
2024-03-14 20:00:07,624 - INFO - run_cmd: sudo -u root setfacl       -m u:user1:-,g:root:-,o::- /tmp/jfs/fsrand/
run_cmd:getfacl /tmp/jfs/fsrand/
2024-03-14 20:00:07,633 - INFO - do_set_acl /tmp/jfs/fsrand/ with u:user1:-,g:root:-,o::- succeed
2024-03-14 20:00:07,634 - INFO - do_chmod /tmp/fsrand/ 0o4 root succeed
2024-03-14 20:00:07,636 - INFO - do_chmod /tmp/jfs/fsrand/ 0o4 root succeed
2024-03-14 20:00:07,637 - INFO - run_cmd: sudo -u user1 setfacl  -R     -m u:root:-,g:root:-,o::- /tmp/fsrand/
2024-03-14 20:00:07,641 - INFO - do_set_acl /tmp/fsrand/ {'user_perm': '-', 'group_perm': '-', 'other_perm': '-'} failed:
setfacl: /tmp/fsrand//aaaa: Permission denied
setfacl: /tmp/fsrand/: Operation not permitted
2024-03-14 20:00:07,641 - INFO - run_cmd: sudo -u user1 setfacl  -R     -m u:root:-,g:root:-,o::- /tmp/jfs/fsrand/
2024-03-14 20:00:07,646 - INFO - do_set_acl /tmp/jfs/fsrand/ {'user_perm': '-', 'group_perm': '-', 'other_perm': '-'} failed:
setfacl: /tmp/jfs/fsrand/: Operation not permitted
setfacl: /tmp/jfs/fsrand/: Permission denied
F
======================================================================
FAIL: test_acl_913 (__main__.TestFsrand2)
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".github/scripts/hypo/fsrand2_test.py", line 44, in test_acl_913
    state.set_acl(default=False, entry=v1, group='root', group_perm=set(), logical=False, mask=set(), not_recalc_mask=False, other_perm=set(), physical=False, recalc_mask=False, recursive=True, set_mask=False, sudo_user='user1', user='root', user_perm=set())
  File "/root/juicefs/.github/scripts/hypo/fsrand2.py", line 418, in set_acl
    target=EntryWithACL,
  File "/root/hypothesis/hypothesis-python/src/hypothesis/stateful.py", line 681, in rule_wrapper
    return f(*args, **kwargs)
  File "/root/juicefs/.github/scripts/hypo/fsrand2.py", line 418, in set_acl
    target=EntryWithACL,
  File "/root/hypothesis/hypothesis-python/src/hypothesis/stateful.py", line 802, in precondition_wrapper
    return f(*args, **kwargs)
  File "/root/juicefs/.github/scripts/hypo/fsrand2.py", line 439, in set_acl
    assert self.equal(result1, result2), f'\033[31mset_acl:\nresult1 is {result1}\nresult2 is {result2}\033[0m'
AssertionError: set_acl:
result1 is
setfacl: /tmp/fsrand//aaaa: Permission denied
setfacl: /tmp/fsrand/: Operation not permitted
result2 is
setfacl: /tmp/jfs/fsrand/: Operation not permitted
setfacl: /tmp/jfs/fsrand/: Permission denied


**What you expected to happen**:
should succeed in user space acl check.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version:
- Cloud provider or hardware configuration running JuiceFS:
- OS (e.g `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Object storage (cloud provider and region, or self maintained):
- Metadata engine info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

More example:
Code:
 ```
   def test_acl_1015(self):
        # SEE: https://github.com/juicedata/jfs/issues/1015
        state = JuicefsMachine()
        v1 = state.init_folders()
        v2 = state.create_file(content=b'', file_name='aaaa', mode='w', parent=v1, umask=0, user='root')
        state.set_acl(default=False, entry=v1, group='root', group_perm={'r'}, logical=False, mask=set(), not_recalc_mask=False, other_perm={'r', 'w', 'x'}, physical=False, recalc_mask=False, recursive=True, set_mask=True, sudo_user='root', user='user1', user_perm={'r', 'w', 'x'})
        state.set_acl(default=False, entry=v1, group='root', group_perm={'r'}, logical=False, mask=set(), not_recalc_mask=False, other_perm=set(), physical=False, recalc_mask=False, recursive=True, set_mask=False, sudo_user='user1', user='root', user_perm=set())
        state.teardown()
```
Log:
```

root@bench-01:~/juicefs# python3 .github/scripts/hypo/fsrand2_test.py -k test_acl_1015
setup_logger ./log1
setup_logger ./log2
__init__
duration is 0.03590893745422363
2024-03-14 20:08:54,861 - INFO - do_create_file /tmp/fsrand/aaaa with mode w succeed
2024-03-14 20:08:54,868 - INFO - do_create_file /tmp/jfs/fsrand/aaaa with mode w succeed
2024-03-14 20:08:54,868 - INFO - run_cmd: sudo -u root setfacl  -R     -m u:user1:wrx,g:root:r,o::wrx,m::- /tmp/fsrand/
run_cmd:getfacl /tmp/fsrand/
2024-03-14 20:08:54,874 - INFO - do_set_acl /tmp/fsrand/ with u:user1:wrx,g:root:r,o::wrx,m::- succeed
2024-03-14 20:08:54,874 - INFO - run_cmd: sudo -u root setfacl  -R     -m u:user1:wrx,g:root:r,o::wrx,m::- /tmp/jfs/fsrand/
run_cmd:getfacl /tmp/jfs/fsrand/
2024-03-14 20:08:54,885 - INFO - do_set_acl /tmp/jfs/fsrand/ with u:user1:wrx,g:root:r,o::wrx,m::- succeed
2024-03-14 20:08:54,885 - INFO - run_cmd: sudo -u user1 setfacl  -R     -m u:root:-,g:root:r,o::- /tmp/fsrand/
2024-03-14 20:08:54,890 - INFO - do_set_acl /tmp/fsrand/ {'user_perm': '-', 'group_perm': 'r', 'other_perm': '-'} failed:
setfacl: /tmp/fsrand//aaaa: Operation not permitted
setfacl: /tmp/fsrand/: Operation not permitted
2024-03-14 20:08:54,890 - INFO - run_cmd: sudo -u user1 setfacl  -R     -m u:root:-,g:root:r,o::- /tmp/jfs/fsrand/
2024-03-14 20:08:54,895 - INFO - do_set_acl /tmp/jfs/fsrand/ {'user_perm': '-', 'group_perm': 'r', 'other_perm': '-'} failed:
setfacl: /tmp/jfs/fsrand/: Operation not permitted
setfacl: /tmp/jfs/fsrand/: Permission denied
F
======================================================================
FAIL: test_acl_1015 (__main__.TestFsrand2)
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".github/scripts/hypo/fsrand2_test.py", line 104, in test_acl_1015
    state.set_acl(default=False, entry=v1, group='root', group_perm={'r'}, logical=False, mask=set(), not_recalc_mask=False, other_perm=set(), physical=False, recalc_mask=False, recursive=True, set_mask=False, sudo_user='user1', user='root', user_perm=set())
  File "/root/juicefs/.github/scripts/hypo/fsrand2.py", line 418, in set_acl
    target=EntryWithACL,
  File "/root/hypothesis/hypothesis-python/src/hypothesis/stateful.py", line 681, in rule_wrapper
    return f(*args, **kwargs)
  File "/root/juicefs/.github/scripts/hypo/fsrand2.py", line 418, in set_acl
    target=EntryWithACL,
  File "/root/hypothesis/hypothesis-python/src/hypothesis/stateful.py", line 802, in precondition_wrapper
    return f(*args, **kwargs)
  File "/root/juicefs/.github/scripts/hypo/fsrand2.py", line 439, in set_acl
    assert self.equal(result1, result2), f'\033[31mset_acl:\nresult1 is {result1}\nresult2 is {result2}\033[0m'
AssertionError: set_acl:
result1 is
setfacl: /tmp/fsrand//aaaa: Operation not permitted
setfacl: /tmp/fsrand/: Operation not permitted
result2 is
setfacl: /tmp/jfs/fsrand/: Operation not permitted
setfacl: /tmp/jfs/fsrand/: Permission denied
```

--- a/pkg/meta/base.go
+++ b/pkg/meta/base.go
@@ -1280,9 +1280,8 @@ func (m *baseMeta) Access(ctx Context, inode Ino, mmask uint8, attr *Attr) sysca
 		}
 	}
 
-	// ref: https://github.com/torvalds/linux/blob/master/fs/namei.c
-	// ci: 5fc475b749c72e0b3f3991b33a90d302f52ae746
-	// dont check acl if group mode is 0
+	// ref: https://github.com/torvalds/linux/blob/e5eb28f6d1afebed4bb7d740a797d0390bd3a357/fs/namei.c#L352-L357
+	// dont check acl if mask is 0
 	if attr.AccessACL != aclAPI.None && (attr.Mode&00070) != 0 {
 		rule := &aclAPI.Rule{}
 		if st := m.en.doGetFacl(ctx, inode, aclAPI.TypeAccess, attr.AccessACL, rule); st != 0 {

--- a/pkg/meta/base.go
+++ b/pkg/meta/base.go
@@ -1280,7 +1280,10 @@ func (m *baseMeta) Access(ctx Context, inode Ino, mmask uint8, attr *Attr) sysca
 		}
 	}
 
-	if attr.AccessACL != aclAPI.None {
+	// ref: https://github.com/torvalds/linux/blob/master/fs/namei.c
+	// ci: 5fc475b749c72e0b3f3991b33a90d302f52ae746
+	// dont check acl if group mode is 0
+	if attr.AccessACL != aclAPI.None && (attr.Mode&00070) != 0 {
 		rule := &aclAPI.Rule{}
 		if st := m.en.doGetFacl(ctx, inode, aclAPI.TypeAccess, attr.AccessACL, rule); st != 0 {
 			return st

--- a/pkg/meta/base_test.go
+++ b/pkg/meta/base_test.go
@@ -165,7 +165,7 @@ func testAccess(t *testing.T, m Meta) {
 	var testNode Ino = 2
 	ctx := NewContext(1, 1, []uint32{2})
 	attr := &Attr{
-		Mode:       0,
+		Mode:       0541,
 		Uid:        0,
 		Gid:        0,
 		AccessACL:  1,

